File: chatwoot/java/src/main/java/oseg/chatwoot_examples/AddNewCustomAttributeToAccountExample.java
package oseg.chatwoot_examples;

import com.chatwoot.client.ApiException;
import com.chatwoot.client.Configuration;
import com.chatwoot.client.api.*;
import com.chatwoot.client.auth.*;
import com.chatwoot.client.JSON;
import com.chatwoot.client.model.*;

import java.io.File;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class AddNewCustomAttributeToAccountExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        config.setApiKey("USER_API_KEY");

        var customAttributeCreateUpdatePayload = new CustomAttributeCreateUpdatePayload();
        customAttributeCreateUpdatePayload.attributeDisplayName(null);
        customAttributeCreateUpdatePayload.attributeDisplayType(null);
        customAttributeCreateUpdatePayload.attributeDescription(null);
        customAttributeCreateUpdatePayload.attributeKey(null);
        customAttributeCreateUpdatePayload.attributeModel(null);
        customAttributeCreateUpdatePayload.attributeValues(List.of ());

        try
        {
            var response = new CustomAttributesApi(config).addNewCustomAttributeToAccount(
                null, // accountId
                customAttributeCreateUpdatePayload // data
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling CustomAttributesApi#addNewCustomAttributeToAccount");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}

File: chatwoot/kotlin/src/main/kotlin/oseg/chatwoot_examples/AddNewCustomAttributeToAccountExample.kt
package oseg.chatwoot_examples

import com.chatwoot.client.infrastructure.*
import com.chatwoot.client.apis.*
import com.chatwoot.client.models.*

import java.io.File
import java.time.LocalDate
import java.time.OffsetDateTime
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.Map
import com.squareup.moshi.adapter

@ExperimentalStdlibApi
class AddNewCustomAttributeToAccountExample
{
    fun addNewCustomAttributeToAccount()
    {
        ApiClient.apiKey["api_access_token"] = "USER_API_KEY"

        val customAttributeCreateUpdatePayload = CustomAttributeCreateUpdatePayload(
            attributeValues = listOf (),
        )

        try
        {
            val response = CustomAttributesApi().addNewCustomAttributeToAccount(
                accountId = 0,
                _data = customAttributeCreateUpdatePayload,
            )

            println(response)
        } catch (e: ClientException) {
            println("4xx response calling CustomAttributesApi#addNewCustomAttributeToAccount")
            e.printStackTrace()
        } catch (e: ServerException) {
            println("5xx response calling CustomAttributesApi#addNewCustomAttributeToAccount")
            e.printStackTrace()
        }
    }
}
